.github/workflows/pr-lint-js.yml

```diff
@@ -0,0 +1,40 @@
+name: Lint JS packages
+
+on:
+    pull_request:
+        paths-ignore:
+            - '**/changelog/**'
+concurrency:
+    group: ${{ github.workflow }}-${{ github.ref }}
+    cancel-in-progress: true
+
+permissions: {}
+
+jobs:
+    lint-test-js:
+        name: Lint
+        runs-on: ubuntu-20.04
+        permissions:
+            contents: read
+            # This is required to allow the action to annotate the PR with the linting results.
+            checks: write
+            pull-requests: read
+        steps:
+            - uses: actions/checkout@v3
+
+            - name: Setup WooCommerce Monorepo
+              uses: ./.github/actions/setup-woocommerce-monorepo
+
+            - name: Lint
+              run: pnpm run -r --filter='release-posts' --filter='woocommerce/client/admin...' --filter='@woocommerce/monorepo-utils' --filter='!@woocommerce/e2e*' --filter='!@woocommerce/api' --color lint
+              continue-on-error: true
+
+            - name: Collect and Combine Eslint Reports
+              run: node ./.github/workflows/scripts/collect-eslint-reports.js
+
+            - name: Annotate Code Linting Results
+              uses: ataylorme/eslint-annotate-action@v2
```

Comment:
Should we pin this to a specific hash?

Reply:
Good call, we should always do that. Thanks for spotting!

Reply:
Resolved in https://github.com/woocommerce/woocommerce/pull/39704/commits/8a197ef967d4ce0346fdc4a3b19f7597de7ad428